ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/load/table/LoadTable.java

```diff
@@ -0,0 +1,199 @@
+  //  private final Helper helper;
+  private final Context context;
+  private final TableContext tableContext;
+  private final TaskTracker tracker;
+  private final TableEvent event;
+
+  public LoadTable(TableEvent event, Context context, TableContext tableContext)
+      throws SemanticException, IOException {
+    this.event = event;
+    this.context = context;
+    this.tableContext = tableContext;
+    this.tracker = new TaskTracker(tableContext.parentTracker);
+  }
+
+  public TaskTracker tasks() throws SemanticException {
+    // Path being passed to us is a table dump location. We go ahead and load it in as needed.
+    // If tblName is null, then we default to the table name specified in _metadata, which is good.
+    // or are both specified, in which case, that's what we are intended to create the new table as.
+    try {
+      if (event.shouldNotReplicate()) {
+        return tracker;
+      }
+      String dbName = tableContext.dbNameToLoadIn; //this can never be null or empty;
+      // Create table associated with the import
+      // Executed if relevant, and used to contain all the other details about the table if not.
+      ImportTableDesc tableDesc = tableContext.overrideProperties(event.tableDesc(dbName));
+      Table table = ImportSemanticAnalyzer.tableIfExists(tableDesc, context.hiveDb);
+      ReplicationSpec replicationSpec = event.replicationSpec();
+
+      // Normally, on import, trying to create a table or a partition in a db that does not yet exist
+      // is a error condition. However, in the case of a REPL LOAD, it is possible that we are trying
+      // to create tasks to create a table inside a db that as-of-now does not exist, but there is
+      // a precursor Task waiting that will create it before this is encountered. Thus, we instantiate
+      // defaults and do not error out in that case.
+      // the above will change now since we are going to split replication load in multiple execution
+      // tasks and hence we could have created the database earlier in which case the waitOnPrecursor will
+      // be false and hence if db Not found we should error out.
+      Database parentDb = context.hiveDb.getDatabase(tableDesc.getDatabaseName());
+      if (parentDb == null) {
+        if (!tableContext.waitOnPrecursor()) {
+          throw new SemanticException(
+              ErrorMsg.DATABASE_NOT_EXISTS.getMsg(tableDesc.getDatabaseName()));
+        }
+      }
+
+      if (table == null) {
+        // If table doesn't exist, allow creating a new one only if the database state is older than the update.
+        if ((parentDb != null) && (!replicationSpec
+            .allowReplacementInto(parentDb.getParameters()))) {
+          // If the target table exists and is newer or same as current update based on repl.last.id, then just noop it.
+          return tracker;
+        }
+      } else {
+        if (!replicationSpec.allowReplacementInto(table.getParameters())) {
```

Comment:
True, but i am hoping that i can reuse code when i change for incremental replication to follow similar limitations as this story. HIVE-17221 has been created  for same.